nautobot_floor_plan/models.py

```diff
@@ -145,8 +184,38 @@ def clean(self):
         - Ensure that the Rack if any belongs to the correct Location.
         - Ensure that this FloorPlanTile doesn't overlap with any other FloorPlanTile in this FloorPlan.
         """
-        # x <= 0, y <= 0 are covered by the base field definitions
         super().clean()
+        FloorPlanTile.allocation_type_assignment(self)
+
+        def group_tile_bounds(rack):
+            """Validate the overlapping of group tiles."""
+            if rack is not None:
+                if x_max > ox_max or x_min < ox_min:
+                    raise ValidationError(
+                        {f"Rack {self.rack} must not extend beyond the boundary of the defined group tiles"}
+                    )
+                if y_max > oy_max or y_min < oy_min:
+                    raise ValidationError(
+                        {f"Rack {self.rack} must not extend beyond the boundary of the defined group tiles"}
+                    )
+                self.on_group_tile = True
+                if orack_group is not None:
+                    if orack_group != rack_group:
+                        # Is tile assigned to a rack_group? Racks must be assigned to the same rack_group
+                        raise ValidationError({"rack_group": f"Rack {self.rack} does not belong to {orack_group}"})
```

Comment:
I talked with DJ on a change on the side, but putting it here for transparency.

If you have a Rack Group tile and you try to assign a tile on top of it with just a Rack (not a Rack Group) it will show up with this error message - even if the Rack in question is indeed in the Rack Group. The current thought is above this we can look to see if they didn't provide the Rack Group in the form and we can just replace it with `rack.rack_group` to "infer" the Rack Group that should have been assigned so we don't throw this error unless the Rack chosen isn't in the proper Rack Group for the larger tile.